### PR TITLE
Remove some unnecessary code from atmsort.pm

### PR DESCRIPTION
**Pull request by hlinnaka**

### CODE REVIEW

File: src/test/regress/atmsort.pm [hunk 1]
```diff
@@ -1112,7 +1112,6 @@ sub atmsort_bigloop
     my $directive = {};
     my $big_ignore = 0;
     my $define_match_expression = undef;
-    my $error_detail_exttab_trifecta_skip = 0; # don't ask!
```
Comment (danielgustafsson) [inline src/test/regress/atmsort.pm]: Any hack with that comment attached deserves to go.. =)

**Comment by danielgustafsson** [comment]
Nice, this PR actually exposed an error in the test output file which had gone unnoticed until now due to the DUMMY replacements:
```diff
diff --git a/src/test/regress/output/external_table.source b/src/test/regress/output/external_table.source
index 6a295fe..75dd12e 100644
--- a/src/test/regress/output/external_table.source
+++ b/src/test/regress/output/external_table.source
@@ -959,7 +959,7 @@ SELECT e1.i, e2.j from exttab_cursor_2 e1 INNER JOIN exttab_cursor_2 e2 ON e1.i
 -- The test is here to track the issue.
 FETCH exttab_cur1;
 ERROR:  Segment reject limit reached. Aborting operation. Last error was: missing data for column "j"  (seg0 slice1 subraa4-mac:40000 pid=70279)
-DETAIL:  External table exttab_cursor_2, line 7 of file://subraa4-mac/Users/abhijits/gitdev/oss_build/gpdb/src/test/regress/data/exttab_more_errors.data: "error_1"
+DETAIL:  External table exttab_cursor_2, line 7 of file://@hostname@@abs_srcdir@/data/exttab_more_errors.data: "error_1"
 COMMIT;
 -- This should have errors populated already
 SELECT count(*) > 0 FROM gp_read_error_log('exttab_cursor_2');
```

**Comment by hlinnaka** [comment]
On 02/12/2017 09:49 AM, Daniel Gustafsson wrote:
> Nice, this PR actually exposed an error in the test output file which had gone unnoticed until now due to the DUMMY replacements:
> ```diff
> diff --git a/src/test/regress/output/external_table.source b/src/test/regress/output/external_table.source
> index 6a295fe..75dd12e 100644
> --- a/src/test/regress/output/external_table.source
> +++ b/src/test/regress/output/external_table.source
> @@ -959,7 +959,7 @@ SELECT e1.i, e2.j from exttab_cursor_2 e1 INNER JOIN exttab_cursor_2 e2 ON e1.i
>  -- The test is here to track the issue.
>  FETCH exttab_cur1;
>  ERROR:  Segment reject limit reached. Aborting operation. Last error was: missing data for column "j"  (seg0 slice1 subraa4-mac:40000 pid=70279)
> -DETAIL:  External table exttab_cursor_2, line 7 of file://subraa4-mac/Users/abhijits/gitdev/oss_build/gpdb/src/test/regress/data/exttab_more_errors.data: "error_1"
> +DETAIL:  External table exttab_cursor_2, line 7 of file://@hostname@@abs_srcdir@/data/exttab_more_errors.data: "error_1"
>  COMMIT;
>  -- This should have errors populated already
>  SELECT count(*) > 0 FROM gp_read_error_log('exttab_cursor_2');
> ```

Hmm, I think we'll still need to suppress that. It can vary, which 
segment the last erroneous row happens to come from, before the limit is 
reached. The line number can also vary, because of that.

I'll add an entry to the  "-- start_matchsubs" block at the top of 
external_table.source for that.

- Heikki

**Comment by danielgustafsson** [comment]
> On 12 Feb 2017, at 09:12, Heikki Linnakangas <notifications@github.com> wrote:
> 
> On 02/12/2017 09:49 AM, Daniel Gustafsson wrote:
> > Nice, this PR actually exposed an error in the test output file which had gone unnoticed until now due to the DUMMY replacements:
> > ```diff
> > diff --git a/src/test/regress/output/external_table.source b/src/test/regress/output/external_table.source
> > index 6a295fe..75dd12e 100644
> > --- a/src/test/regress/output/external_table.source
> > +++ b/src/test/regress/output/external_table.source
> > @@ -959,7 +959,7 @@ SELECT e1.i, e2.j from exttab_cursor_2 e1 INNER JOIN exttab_cursor_2 e2 ON e1.i
> > -- The test is here to track the issue.
> > FETCH exttab_cur1;
> > ERROR: Segment reject limit reached. Aborting operation. Last error was: missing data for column "j" (seg0 slice1 subraa4-mac:40000 pid=70279)
> > -DETAIL: External table exttab_cursor_2, line 7 of file://subraa4-mac/Users/abhijits/gitdev/oss_build/gpdb/src/test/regress/data/exttab_more_errors.data: "error_1"
> > +DETAIL: External table exttab_cursor_2, line 7 of file://@hostname@@abs_srcdir@/data/exttab_more_errors.data: "error_1"
> > COMMIT;
> > -- This should have errors populated already
> > SELECT count(*) > 0 FROM gp_read_error_log('exttab_cursor_2');
> > ```
> 
> Hmm, I think we'll still need to suppress that. It can vary, which 
> segment the last erroneous row happens to come from, before the limit is 
> reached. The line number can also vary, because of that.
> 
> I'll add an entry to the "-- start_matchsubs" block at the top of 
> external_table.source for that.

With a segment reject limit of 2, the line number reported should be stable I
would think given how the external file would be processed.  The preceeding
comment in the file points out an existing bug though, missed that comment
earlier when reading.

**Comment by hlinnaka** [comment]
On 02/12/2017 10:22 AM, Daniel Gustafsson wrote:
>
>> On 12 Feb 2017, at 09:12, Heikki Linnakangas <notifications@github.com> wrote:
>>
>> On 02/12/2017 09:49 AM, Daniel Gustafsson wrote:
>>> Nice, this PR actually exposed an error in the test output file which had gone unnoticed until now due to the DUMMY replacements:
>>> ```diff
>>> diff --git a/src/test/regress/output/external_table.source b/src/test/regress/output/external_table.source
>>> index 6a295fe..75dd12e 100644
>>> --- a/src/test/regress/output/external_table.source
>>> +++ b/src/test/regress/output/external_table.source
>>> @@ -959,7 +959,7 @@ SELECT e1.i, e2.j from exttab_cursor_2 e1 INNER JOIN exttab_cursor_2 e2 ON e1.i
>>> -- The test is here to track the issue.
>>> FETCH exttab_cur1;
>>> ERROR: Segment reject limit reached. Aborting operation. Last error was: missing data for column "j" (seg0 slice1 subraa4-mac:40000 pid=70279)
>>> -DETAIL: External table exttab_cursor_2, line 7 of file://subraa4-mac/Users/abhijits/gitdev/oss_build/gpdb/src/test/regress/data/exttab_more_errors.data: "error_1"
>>> +DETAIL: External table exttab_cursor_2, line 7 of file://@hostname@@abs_srcdir@/data/exttab_more_errors.data: "error_1"
>>> COMMIT;
>>> -- This should have errors populated already
>>> SELECT count(*) > 0 FROM gp_read_error_log('exttab_cursor_2');
>>> ```
>>
>> Hmm, I think we'll still need to suppress that. It can vary, which
>> segment the last erroneous row happens to come from, before the limit is
>> reached. The line number can also vary, because of that.
>>
>> I'll add an entry to the "-- start_matchsubs" block at the top of
>> external_table.source for that.
>
> With a segment reject limit of 2, the line number reported should be stable I
> would think given how the external file would be processed.

I think you're right. I don't quite understand how that works out, I 
would assume that it depends on the order that the master receives the 
error results from the segments. But there are a lot of similar queries 
using the same external table, and they all report the error at line 7. 
So I guess it's OK.

I simply updated the expected output, without adding any new 
matchsubs-rules. Let's see if the pipeline likes it now. If the output 
turns out to be unstable, we can fix it then.

- Heikki

**Comment by hlinnaka** [comment]
Pushed, thanks for the review.

**Pull request closed without merging**